tests/tests_integration/test_api/test_iam.py

```diff
@@ -59,12 +59,15 @@ def test_create_and_delete(self, cognite_client):
         assert res.id not in {s.id for s in cognite_client.iam.security_categories.list()}
 
 
+@pytest.mark.skipif(
+    os.getenv("LOGIN_FLOW") == "client_certificate", reason="Sessions do not work with client_certificate"
+)
 class TestSessionsAPI:
-    @pytest.mark.skip("Bad test: CogniteAPIError: There can be only 10000 sessions")
-    @pytest.mark.skipif(
-        os.getenv("LOGIN_FLOW") == "client_certificate", reason="Sessions do not work with client_certificate"
-    )
-    def test_create_and_revoke(self, cognite_client):
+    def test_create_retrieve_and_revoke(self, cognite_client: CogniteClient) -> None:
         res = cognite_client.iam.sessions.create()
+
+        retrieved = cognite_client.iam.sessions.retrieve(res.id)
+
+        assert retrieved.id == res.id
         assert res.id in {s.id for s in cognite_client.iam.sessions.list("READY")}
         assert res.id in {s.id for s in cognite_client.iam.sessions.revoke(res.id) if s.status == "REVOKED"}
```

Comment:
Line 72 and 73 feels like mixtures of setup, teardown and asserts 😅 Consider if its worth an improvement (idea: should we revoke session in a `finally` clause?)

Reply:
What about now? Also discovered that session.revoked was not overloaded correctly.

Reply:
Nice 👌